crates/telio-proxy/src/proxy.rs

```diff
@@ -283,27 +318,48 @@ impl StateEgress {
     }
 
     async fn send_outbound_data(&mut self, pk: PublicKey, msg: DataMsg) {
-        // if let Some((pk, msg)) = self.input.recv().await {
-        if let (Some((socket, None)), Some(wg_addr)) = (self.sockets.get(&pk), self.wg_addr) {
-            match socket.send_to(msg.get_payload(), wg_addr).await {
-                Ok(_) => {
-                    if self.conn_state.is_err() {
-                        self.conn_state = Ok(());
-                        telio_log_info!("Outbound conn telio -> WG working!");
+        if let Err(err) = self.send_data(pk, msg.clone()).await {
+            let mut should_resend = matches!(
```

Comment:
NIT: you could probably just add method to SocketResentBehaviour like ::should_resend(), reutining bool, stead of doing matches multiple times.